### DISCUSSION TITLE
How to use pnp strict with workspaces? And how to handle typescript with references?

### DISCUSSION THREAD

**Discussion by ikibalnyi** [yarnpkg/berry#2246] [General]
1. How can I run pnpify with `pnpMode: strict` in workspaces?
For example I need to run `tsc` on one of the packages:
I try `yarn workspace innerPackage pnpify tsc`. And fails with error `Usage Error: Couldn't find a script named "pnpify"`.
Should I install `@yarnpkg/pnpify` for each package or is there any alternatives?

2. When I add `@yarnpkg/pnpify` to the inner package and run `yarn pnpify tsc -b`. It's not able to resolve `react` in referenced modules and fails with:
```
TS2307: Cannot find module 'react' or its corresponding type declarations.

1 import type { ComponentProps, ReactElement, ReactNode, ComponentType } from 'react';
```
Same as I would run it without pnpfy.

**Comment by merceyz** [comment]
> How can I run pnpify with pnpMode: strict in workspaces?

It does that by default but the point of `pnpify` is to emulate node_modules in memory which by definition isn't strict

> Should I install @yarnpkg/pnpify for each package or is there any alternatives?

Yes, but you don't need pnpify for `tsc`

> When I add @yarnpkg/pnpify to the inner package and run yarn pnpify tsc -b. It's not able to resolve react in referenced modules and fails with:

I'd guess you lack a dependency on `@types/react` but without a reproduction it's hard to tell

**Reply by ikibalnyi** [reply]
Nope, react is present in the referenced package as well as in the package where I run the command from.
